net: config: build break when CONFIG_NET_NATIVE=n

**Describe the bug**
When `CONFIG_NET_NATIVE=n` is set and building any of the networking samples, the build will break with the following message:
```
zephyr-master/zephyr/subsys/net/lib/config/init.c: In function 'check_interface':
zephyr/subsys/net/lib/config/init.c:323:9: warning: 'return' with a value, in function returning void [-Wreturn-type]
  323 |  return true;
      |         ^~~~
zephyr/subsys/net/lib/config/init.c:318:13: note: declared here
  318 | static void check_interface(struct net_if *iface)
      |             ^~~~~~~~~~~~~~~
zephyr/subsys/net/lib/config/init.c: In function 'net_config_init':
zephyr/subsys/net/lib/config/init.c:352:6: error: void value not ignored as it ought to be
  352 |  if (check_interface(iface) == false) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

It looks like a refactoring of `check_interface` in `subsys/net/lib/config/init.c` missed the return value change to `bool`.

**To Reproduce**
Found bug when testing NET_NATIVE=n with SARA-R4 modem using LwM2M client::
```
1. west build -b frdm_k64f -s zephyr/samples/net/lwm2m_client -- \
  -DSHIELD=sparkfun_sara_r4 \
  -DCONFIG_NET_CONFIG_PEER_IPV4_ADDR=\"23.97.187.154\" \
  -DCONFIG_NET_IPV6=n -DCONFIG_NET_CONFIG_NEED_IPV6=n \
  -DCONFIG_MODEM_UBLOX_SARA_R4_APN=\"<MODEM_APN>\" \
  -DCONFIG_MODEM_UBLOX_SARA_R4_MANUAL_MCCMNO=\"<MODEM_MCCMNO>\" \
  -DCONFIG_COAP_INIT_ACK_TIMEOUT_MS=10000 -DCONFIG_MODEM_SHELL=y \
  -DCONFIG_LWM2M_SERVER_DEFAULT_PMIN=1 -DCONFIG_LWM2M_SERVER_DEFAULT_PMAX=360 \
  -DCONFIG_LOG_STRDUP_MAX_STRING=128 \
  -DCONFIG_LOG_STRDUP_BUF_COUNT=10 -DCONFIG_NET_NATIVE=n
2. See error
```

**Expected behavior**
Build should not break.

**Impact**
NET_NATIVE is currently broken

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Toolchain: Zephyr SDK 0.11.1
 - Commit SHA: Todays master (5a7f073db64b8c1ffa0300eebfb39a0acb70fae4)

--- a/subsys/net/lib/config/init.c
+++ b/subsys/net/lib/config/init.c
@@ -315,7 +315,7 @@ static bool check_interface(struct net_if *iface)
 	return false;
 }
 #else
-static void check_interface(struct net_if *iface)
+static bool check_interface(struct net_if *iface)
 {
 	k_sem_reset(&counter);
 	k_sem_give(&waiter);

